tests/purs/failing/FoldableInstance5.out

```diff
@@ -0,0 +1,10 @@
+Error found:
+in module [33mFoldableInstance5[0m
+at tests/purs/failing/FoldableInstance5.purs:9:1 - 9:30 (line 9, column 1 - line 9, column 30)
+
+  Type constructor ‘Test' must use the type variable 'a' only as the last argument of a data type
+
+
+See https://github.com/purescript/documentation/blob/master/errors/CannotDeriveInvalidTypeConstructor.md for more information,
+or to contribute content related to this error.
+
```

Comment:
Should there should be a hint here that clarifies that we were trying to derive an instance for `Foldable` when this error was encountered? The only such hint provided is the source spans.

Reply:
Yeah, we should include the class. We might want to put the same check in `Functor` and `Traversable`, even though it isn't strictly necessary, because the message could be more helpful than a type mismatch error for code that the user never wrote.